app/src/copymultiplekeyframesdialog.cpp

```diff
@@ -0,0 +1,276 @@
+{
+    return ui->sBoxStartReverse->value();
+}
+
+bool CopyMultiplekeyframesDialog::getValidity()
+{
+    return mValidAction;
+}
+
+void CopyMultiplekeyframesDialog::setFirstFrame(int firstFrame)
+{
+    mFirstFrame = firstFrame;
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setLastFrame(int lastFrame)
+{
+    mLastFrame = lastFrame;
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setFromLayer(QString fromLayer)
+{
+    mFromLayer = fromLayer;
+    int index = ui->tabWidget->currentIndex();
+    switch (index) {
+    case 0:
+        ui->labInfoToFromLayer->setText(tr("From: %1 To: %2").arg(ui->cBoxFromLayer->currentText()).arg(ui->cBoxCopyToLayer->currentText()));
+        break;
+    case 1:
+        ui->labInfoToFromLayer->setText(tr("From: %1 To: %2").arg(ui->cBoxFromLayer->currentText()).arg(ui->cBoxMoveToLayer->currentText()));
+        break;
+    case 2:
+        ui->labInfoToFromLayer->setText(tr("On: %1").arg(mFromLayer));
+        break;
+    case 3:
+        ui->labInfoToFromLayer->setText(tr("On: %1").arg(mFromLayer));
+        ui->labDeleteOnLayer->setText(tr("On Layer %1").arg(mFromLayer));
+        break;
+    default:
+        Q_ASSERT(false);
+        break;
+    }
+}
+
+void CopyMultiplekeyframesDialog::setCopyToLayer(QString copyToLayer)
+{
+    mCopyToLayer = copyToLayer;
+    ui->labInfoToFromLayer->setText(tr("From: %1 To: %2").arg(ui->cBoxFromLayer->currentText()).arg(ui->cBoxCopyToLayer->currentText()));
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setNumLoops(int numLoops)
+{
+    mNumLoops = numLoops;
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setStartFrame(int startFrame)
+{
+    mCopyStart = startFrame;
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setMoveStartFrame(int startFrame)
+{
+    mMoveStart = startFrame;
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setMoveToLayer(QString moveToLayer)
+{
+    mMoveToLayer = moveToLayer;
+    ui->labInfoToFromLayer->setText(tr("From: %1 To: %2").arg(ui->cBoxFromLayer->currentText()).arg(ui->cBoxMoveToLayer->currentText()));
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setReverseFrom(int reverseFrom)
+{
+    mReverseStart = reverseFrom;
+    ui->labInfoToFromLayer->setText(tr("On Layer: %1").arg(ui->cBoxFromLayer->currentText()));
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setDeleteOnLayer(QString deleteFromLayer)
+{
+    mDeleteOnLayer = deleteFromLayer;
+    ui->labInfoToFromLayer->setText(tr("On: %1").arg(mDeleteOnLayer));
+    ui->labDeleteOnLayer->setText(tr("On Layer %1").arg(mDeleteOnLayer));
+    checkValidity();
+}
+
+void CopyMultiplekeyframesDialog::setMethodPicked(int tabIndex)
+{
+    mCurrentTab = tabIndex;
+    ui->labInfoAction->setText(ui->tabWidget->tabText(mCurrentTab));
+    checkValidity();
+}
+
+
+void CopyMultiplekeyframesDialog::setStartEnd(int methodChosen)
+{
+    switch (methodChosen) {
+    case 0: // copy
```

Comment:
I can see the logic behind your request @CandyFace , but the parameter methodChosen refers to the tab index. I only use the method twice, but first time I use it (in 'init()') I want tab 0 (Copy) to be chosen, since that probable will be the most used of the four. 
It works as it is now, and I'd like to keep it. What do you say?